superset/db_engine_specs/questdb.py

```diff
@@ -0,0 +1,311 @@
+from superset.db_engine_specs.base import (
+    BaseEngineSpec,
+    BasicParametersMixin,
+    BasicParametersType,
+    TimeGrain,
+    builtin_time_grains,
+)
+from superset.utils import core as utils
+from superset.utils.core import GenericDataType
+
+from questdb_connect import remove_public_schema, types
+from questdb_connect.dialect import connection_uri
+from questdb_connect.function_names import FUNCTION_NAMES
+
+# Apache Superset requires a Python DB-API database driver, and a SQLAlchemy dialect
+# https://superset.apache.org/docs/databases/installing-database-drivers
+# https://preset.io/blog/building-database-connector/
+# https://preset.io/blog/improving-apache-superset-integration-database-sqlalchemy/
+
+
+class QDBParametersSchema(Schema):
+    username = fields.String(allow_none=True, description=__('user'))
+    password = fields.String(allow_none=True, description=__('password'))
+    host = fields.String(required=True, description=__('host'))
+    port = fields.Integer(allow_none=True, description=__('port'))
+    database = fields.String(allow_none=True, description=__('database'))
+
+
+class QDBEngineSpec(BaseEngineSpec, BasicParametersMixin):
+    engine = 'questdb'
+    engine_name = 'QuestDB Connect'
+    default_driver = "psycopg2"
+    encryption_parameters = {"sslmode": "prefer"}
+    sqlalchemy_uri_placeholder = "questdb://user:password@host:port/database"
+    parameters_schema = QDBParametersSchema()
+    time_groupby_inline = False
+    allows_hidden_cc_in_orderby = True
+    time_secondary_columns = True
+    try_remove_schema_from_table_name = True
+    max_column_name_length = 120
+    supports_dynamic_schema = False
+    top_keywords = {}
+    # https://en.wikipedia.org/wiki/ISO_8601#Durations
+    # https://questdb.io/docs/reference/function/date-time/#date_trunc
+    _time_grain_expressions = {
+        None: '{col}',
+        "PT1S": "date_trunc('second', {col})",
+        "PT5S": "date_trunc('second', {col}) + 5000000L",
+        "PT30S": "date_trunc('second', {col}) + 30000000L",
+        "PT1M": "date_trunc('minute', {col})",
+        "PT5M": "date_trunc('minute', {col}) + 300000000L",
+        "PT10M": "date_trunc('minute', {col}) + 600000000L",
+        "PT15M": "date_trunc('minute', {col}) + 900000000L",
+        "PT30M": "date_trunc('minute', {col}) + 1800000000L",
+        "PT1H": "date_trunc('hour', {col})",
+        "PT6H": "date_trunc('hour', {col}) + 21600000000L",
+        "P1D": "date_trunc('day', {col})",
+        "P1W": "date_trunc('week', {col})",
+        "P1M": "date_trunc('month', {col})",
+        "P1Y": "date_trunc('year', {col})",
+        "P3M": "date_trunc('quarter', {col})",
+    }
+    ret_list = []
+    for duration, func in _time_grain_expressions.items():
+        if duration:
+            name = builtin_time_grains[duration]
+            ret_list.append(TimeGrain(name, _(name), func, duration))
+    _engine_time_grains = tuple(ret_list)
+    _default_column_type_mappings = (
```

Comment:
`_default_column_type_mappings` should not be overridden - please define `column_type_mappings` instead

Reply:
thank you, resolved in 9a3c588dc495ad0435666f41b455d54ee57b40fc